src/jvm/main/org/jetbrains/kotlinx/lincheck/trace/TraceCompressor.kt

```diff
@@ -21,6 +21,20 @@ internal fun SingleThreadedTable<TraceNode>.compressTrace() = this
     .compressUserThreadRun()
     .compressThreadStart()
     .removeCoroutinesCoreSuffix()
+    .compressInlineIV()
+    .compressDollarThis()
+    .replaceNestedClassDollar()
+    .compressSyntheticParameterNumbers()
+    .compressLambdaCaptureSyntheticField()
+    .compressVolatileDollar()
```

Comment:
For the transformations that alter only the `ownerName` of the trace points, I have alternative suggestions on how to approach them. Let's consider them case-by-case to find the best option. See comments below

Reply:
see https://github.com/JetBrains/lincheck/pull/683#issuecomment-2983300439